run_app.py

```diff
@@ -1,7 +1,38 @@
-from streamlit.web import cli
+import argparse
+import json
+import subprocess
+import logging
+
+# Configure logging
+logging.basicConfig(level=logging.WARNING)
+
+# Default upload size (MB) if settings.json is missing/invalid
+DEFAULT_MAX_SIZE = 200
+
+def get_upload_limit():
+    """Read max upload size from settings.json or use default."""
+    try:
+        with open("settings.json") as f:
+            return json.load(f).get("maximum_file_upload_size", DEFAULT_MAX_SIZE)
+    except (FileNotFoundError, json.JSONDecodeError):
+        return DEFAULT_MAX_SIZE
 
 if __name__ == "__main__":
-    cli._main_run_clExplicit(
-        file="app.py", command_line="streamlit run"
-    )
-    # we will create this function inside our streamlit framework
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--max-upload', type=int, help='Override upload size (MB)')
+    args, unknown = parser.parse_known_args()
```

Comment:
I am not sure if passing this via argparser is appropriate here, might be too confusing for a user as all the other parameters are set with settings.json except for this one. I would just stick with keeping all the settings in settings.json